### PR TITLE
feat(ingest): add python utility classes for NotebookUrn, CorpuserUrn and CorpGroupUrn

### PR DESCRIPTION
**Pull request by tc350981**
## Checklist
- [x] The PR conforms to DataHub's [Contributing Guideline](https://github.com/datahub-project/datahub/blob/master/docs/CONTRIBUTING.md) (particularly [Commit Message Format](https://github.com/datahub-project/datahub/blob/master/docs/CONTRIBUTING.md#commit-message-format))
- [ ] Links to related issues (if applicable)
- [ ] Tests for the changes have been added/updated (if applicable)
- [ ] Docs related to the changes have been added/updated (if applicable)

### CODE REVIEW

**Comment by tc350981** [comment]
the smoke test failure seems not related to my change

File: metadata-ingestion/src/datahub/utilities/urns/corpuser_urn.py [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+
+
+class CorpuserUrn(Urn):
+    """
+    expected corp user urn format: urn:li:corpuser:<corpuser_id>. example: "urn:li:corpuser:tom"
+    """
+
+    ENTITY_TYPE: str = "corpuser"
+
+    def __init__(self, entity_type: str, entity_id: List[str], domain: str = "li"):
```
Comment (rslanka) [inline metadata-ingestion/src/datahub/utilities/urns/corpuser_urn.py:14]:
```suggestion
    def __init__(self, entity_type: str, entity_id: List[str], domain: str = Urn.LI_DOMAIN):
```

File: metadata-ingestion/src/datahub/utilities/urns/notebook_urn.py [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+
+
+class NotebookUrn(Urn):
+    """
+    expected dataset urn format: urn:li:notebook:(<platform_name>,<notebook_id>). example: "urn:li:notebook:(querybook,1234)"
+    """
+
+    ENTITY_TYPE: str = "notebook"
+
+    def __init__(self, entity_type: str, entity_id: List[str], domain: str = "li"):
```
Comment (rslanka) [inline metadata-ingestion/src/datahub/utilities/urns/notebook_urn.py:14]:
```suggestion
    def __init__(self, entity_type: str, entity_id: List[str], domain: str = Urn.LI_DOMAIN):
```

File: metadata-ingestion/src/datahub/utilities/urns/notebook_urn.py [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+        super().__init__(entity_type, entity_id, domain)
+
+    @classmethod
+    def create_from_string(cls, urn_str: str) -> "NotebookUrn":
+        urn: Urn = super().create_from_string(urn_str)
+        return cls(urn.get_type(), urn.get_entity_id(), urn.get_domain())
+
+    @classmethod
+    def create_from_id(cls, platform_id: str, notebook_id: str) -> "NotebookUrn":
```
Comment (rslanka) [inline metadata-ingestion/src/datahub/utilities/urns/notebook_urn.py:23]:
```suggestion
    def create_from_ids(cls, platform_id: str, notebook_id: str) -> "NotebookUrn":
```

**Review by rslanka — COMMENTED** [review]
Looks great! @tc350981, please review the suggestions and we can merge this.

**Comment by rslanka** [comment]
Hi @tc350981, my suggestions might have introduced some linter issues. Could you please fix them?

**Review by rslanka — APPROVED** [review]
LGTM!

**Pull request merged**